pkg/volumes/do/volumes.go

```diff
@@ -136,54 +155,136 @@ func (a *DOVolumes) findVolumes(filterByRegion bool) ([]*volumes.Volume, error)
-		if strings.Contains(doVolume.Name, "etcd-main") {
-			clusterKey = "main"
-		} else if strings.Contains(doVolume.Name, "etcd-events") {
-			clusterKey = "events"
-		} else {
-			clusterKey = ""
-			glog.V(2).Infof("could not determine etcd cluster type for volume: %s", doVolume.Name)
+		for _, dropletTag := range a.dropletTags {
+			glog.V(2).Infof("Iterating all droplet tags for droplet name=%s dropletTag=%s", a.dropletName, dropletTag)
 		}
 
-		if strings.Contains(a.nameTag, clusterKey) {
-			glog.V(2).Infof("Found a matching nameTag=%s for cluster key=%s with etcd cluster name = %s and k8s cluster name=%s", a.nameTag, clusterKey, a.ClusterName, k8sclusterName)
-			vol := &volumes.Volume{
-				ProviderID: doVolume.ID,
-				Info: volumes.VolumeInfo{
-					Description: a.ClusterName + "-" + a.nameTag,
-				},
-				MountName: "master-" + doVolume.ID,
-				EtcdName:  a.ClusterName + "-" + clusterKey,
+		// make sure dropletTags match the volumeTags for the keys mentioned in matchTags
+		tagFound := a.matchesTags(&doVolume)
+
+		if tagFound {
+
+			glog.V(2).Infof("Tag Matched for droplet name=%s and volume name=%s", a.dropletName, doVolume.Name)
+
+			var clusterKey string
+			if strings.Contains(doVolume.Name, "etcd-main") {
+				clusterKey = "main"
+			} else if strings.Contains(doVolume.Name, "etcd-events") {
+				clusterKey = "events"
+			} else {
+				clusterKey = ""
+				glog.V(2).Infof("could not determine etcd cluster type for volume: %s", doVolume.Name)
 			}
 
-			if len(doVolume.DropletIDs) == 1 {
-				vol.AttachedTo = strconv.Itoa(doVolume.DropletIDs[0])
-				vol.LocalDevice = getLocalDeviceName(&doVolume)
+			if strings.Contains(a.nameTag, clusterKey) {
+				vol := &volumes.Volume{
+					ProviderID: doVolume.ID,
+					Info: volumes.VolumeInfo{
+						Description: a.ClusterName + "-" + a.nameTag,
+					},
+					MountName: "master-" + doVolume.ID,
+					EtcdName:  doVolume.Name,
+				}
+
+				if len(doVolume.DropletIDs) == 1 {
+					vol.AttachedTo = strconv.Itoa(doVolume.DropletIDs[0])
+					vol.LocalDevice = getLocalDeviceName(&doVolume)
+				}
+
+				glog.V(2).Infof("Found a matching nameTag=%s for cluster key=%s with etcd cluster name = %s; volume name = %s", a.nameTag, clusterKey, a.ClusterName, doVolume.Name)
+
+				myvolumes = append(myvolumes, vol)
 			}
+		}
+	}
+
+	return myvolumes, nil
+}
+
+func (a *DOVolumes) matchesTags(volume *godo.Volume) bool {
+	for k, v := range a.matchTags {
+		// In DO, we don't have tags with key value pairs. So we are storing them separating by a ":"
+		// Create a string like "k:v" and check if they match.
+		doTag := k + ":" + v
 
-			glog.V(2).Infof("Matching DO Volume found with name=%s ID=%s etcdname=%s mountname=%s", doVolume.Name, doVolume.ID, vol.EtcdName, vol.MountName)
+		glog.V(2).Infof("Check for matching volume tag - doTag=%s for volume name = %s", strings.ToUpper(doTag), volume.Name)
 
-			myvolumes = append(myvolumes, vol)
+		volumeTagfound := a.Contains(volume.Tags, doTag)
```

Comment:
Nit: volumeTagFound

You might also consider making volume.Tags into a map (`map[string]bool`) or splitting the tags when you first see them `map[string]string`

Reply:
I'll keep it this way for now, and consider an update in the next pass.